src/modes.cpp

```diff
@@ -231,6 +243,21 @@ void aste::runMapperMode(const aste::ExecutionContext &context, const OptionMap
             break;
           }
         }
+#ifdef ASTE_NN_GRADIENT_MAPPING
+        else if (meshdata.type == aste::datatype::GRADIENT) {
+          switch (meshdata.numcomp) {
+          case 1:
+            assert(meshdata.dataVector.size() == vertexIDs.size() * dim);
+            // preciceInterface.writeBlockScalarData(meshdata.dataID, vertexIDs.size(), vertexIDs.data(), meshdata.dataVector.data());
+            preciceInterface.writeBlockScalarGradientData(meshdata.dataID, vertexIDs.size(), vertexIDs.data(), meshdata.dataVector.data());
+            break;
+          default:
+            assert(meshdata.dataVector.size() == vertexIDs.size() * dim * dim);
+            preciceInterface.writeBlockVectorGradientData(meshdata.dataID, vertexIDs.size(), vertexIDs.data(), meshdata.dataVector.data(), true);
+            break;
```

Comment:
Can you please double check that the gradient data is passed into the function according to the documentation here https://github.com/precice/precice/blob/eecad458848d47889c69c7ca64191891dac45870/src/precice/SolverInterface.hpp#L1100-L1113
We modified this later on, so the website docs might not yet been synced and detecting this later on would be very hard. You could also run simple test cases with functions having non-zero derivatives in particular directions.

Reply:
Row first manner is adopted all x derivatives all y derivates and all z derivates